Source/Editor/DualityEditor/Utility/SerializableReferenceWrapper.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Duality.Editor
+{
+	/// <summary>
+	/// A wrapper object that stores a reference to a non-<see cref="SerializableAttribute"/> object.
+	/// </summary>
+	[Serializable]
+	public class SerializableReferenceWrapper : SerializableWrapper
+	{
+		private static long nextID = 0;
+		private static readonly Dictionary<long, object> referenceMap 
+			= new Dictionary<long, object>();
+
+		private long ID = -1;
+
+		public sealed override object Data
+		{
+			get
+			{
+				return base.Data;
+			}
+			set
+			{
+				// Invalidate the ID in use. Could
+				// occur if this SerializableReferenceWrapper
+				// is being reused
+				this.ID = -1;
+				base.Data = value;
+			}
+		}
+
+		public SerializableReferenceWrapper(object data)
+			: base(data)
+		{
+		}
+		private SerializableReferenceWrapper(SerializationInfo info, StreamingContext context)
+		{
+			object serializedObject = info.GetValue("data", typeof(long));
+			if (serializedObject is long)
+			{
+				this.ID = (long) serializedObject;
+				base.Data = referenceMap[this.ID];
+			}
+			else
+			{
+				this.ID = -1;
+				base.Data = null;
```

Comment:
`base.XY` on a property used anywhere else than inside the property override itself is a bit of an unusual pattern, and I've had to think about what would happen. Assuming that's not just me, maybe just make the field `protected` and use it directly.

It would only be a minor change though, can address this myself during merge.